src/components/search/data/searchLoader.test.jsx

```diff
@@ -124,4 +143,74 @@ describe('searchLoader', () => {
       }),
     );
   });
+
+  it('Redirect learners whose enterprise has enabled one academy.', async () => {
+    extractEnterpriseCustomer.mockResolvedValue(enterpriseCustomerFactory({ enable_one_academy: true }));
+    const academiesQuery = queryAcademiesList(mockEnterpriseCustomer.uuid);
+
+    when(mockQueryClient.ensureQueryData).calledWith(
+      expect.objectContaining({
+        queryKey: academiesQuery.queryKey,
+      }),
+    ).mockResolvedValue(mockAcademies);
+    ensureAuthenticatedUser.mockResolvedValue({ userId: 3, username: 'test-user' });
+
+    const mockParams = { enterpriseSlug: mockEnterpriseCustomer.slug };
+    const searchLoader = ({ request }) => makeSearchLoader(mockQueryClient)({ params: mockParams, request });
+
+    renderWithRouterProvider({
+      path: `/${mockEnterpriseCustomer.slug}/search`,
+      element: <div data-testid="root" />,
+      loader: searchLoader,
+    }, {
+      routes: [
+        {
+          path: `/${mockEnterpriseCustomer.slug}/academies/${mockAcademies[0].uuid}`,
+          element: <div data-testid="root" />,
+        },
+      ],
+      initialEntries: [`/${mockEnterpriseCustomer.slug}/search`],
+    });
+
+    await waitFor(() => {
+      expect(screen.getByTestId('root')).toBeInTheDocument();
+    });
+
+    expect(redirect).toHaveBeenCalledWith(`/${mockEnterpriseCustomer.slug}/academies/${mockAcademies[0].uuid}`);
```

Comment:
Rather than waiting for the `root` element to render and asserting on a mocked `redirect` function, the test should assert that the academy detail specific `data-testid` is rendered (aka did React Router properly `redirect` to the new route?).

```jsx
expect(screen.getByTestId('academy-detail').toBeInTheDocument();
```